feat(entityHierarchy): RN-1853: Migrate entities to projects (#6749)
beyondessential/tupaia: llm/specs/RN-1853-refinement.md

- **`project_country` junction table.** Only needed once `entity_parent_child_relation` is removed. Belongs with [RN-1864](https://linear.app/bes/issue/RN-1864/remove-entity-relation-and-entity-hierarchy-tables).
- **Unbundling `map_overlay.country_codes` triple-role** (scoping vs permission vs project indirection). Juliana flagged this; not required under this approach.
- **Removing `entity_parent_child_relation`** — [RN-1862](https://linear.app/bes/issue/RN-1862/consolidate-hierarchy-to-parent-id-on-project-specific-entities).
- **Removing `entity_hierarchy` and `entity_relation` tables** — [RN-1864](https://linear.app/bes/issue/RN-1864/remove-entity-relation-and-entity-hierarchy-tables).
- **Project-scoping all entity queries** — [RN-1854](https://linear.app/bes/issue/RN-1854/ensure-all-entity-access-is-project-scoped). Bleeds into RN-1853 only where callsites break; the bulk is its own ticket.

---

## Risks / things to watch

> - **`entity.code` is no longer unique post-migration.** Many callsites assume `findOne({ code })` returns at most one row. Will need a codebase audit (mostly RN-1854 territory).

Reviewer: This point is subtly incorrect: It’s fine to assume `findOne` returns either `null` or one record; it will no longer be safe to assume that it’s the **expected** record.

```suggestion
- **`entity.code` is no longer unique post-migration.** Will need a codebase audit  of `findOne({ code })` usages (mostly RN-1854 territory).
```
